### DISCUSSION TITLE
Possible to specify a dependency's dependency?

### DISCUSSION THREAD

**Discussion by ricotrevisan** [esm-dev/esm.sh#845] [Q&A]
I'm trying to load and run Tiptap and I get an error that one of its (many) dependencies was loaded multiple times.
(I recognize that is likely a problem for Tiptap to fix (it also happens with Skypack) -- I am hoping for a workaround
FYI: my post on [Tiptap's forum](https://github.com/ueberdosis/tiptap/discussions/2529#discussioncomment-9782317)) 

![CleanShot 2024-06-15 at 16 29 45@2x](https://github.com/esm-dev/esm.sh/assets/2459327/c1e17d0e-f95d-4c37-82b7-418ee2481d07)

# Network tab
Looking at the network tab, I can see that `prosemirror-model` was indeed loaded multiple times

![CleanShot 2024-06-15 at 16 31 16@2x](https://github.com/esm-dev/esm.sh/assets/2459327/d6983012-4997-4c5a-9070-8aef000ff5a5)

# Inspecting
And they load 2 different versions

![CleanShot 2024-06-15 at 16 36 14@2x](https://github.com/esm-dev/esm.sh/assets/2459327/3c4d8e1f-f7ca-4a21-81f1-0ee5ce66a62c)

---

# Steps to reproduce
1. open this Codepen: https://codepen.io/ricotrevisan/pen/poqambQ
2. open the browser's console (not Codepen's console)
3. reload the page
4. in the page, after "Example Text", hit enter
5. the error shows up in the browser's console (not Codepen's console).

**Comment by ije** [comment]
i don't think you can it now. you can use import maps to make the dependency as an exteranl module, check https://preactjs.com/guide/v10/getting-started#using-preact-with-htm-and-importmaps
next version fo esm.sh going to use **semver caret(e.g. ^1.0.0) range** for dependecies instead of current fixed version, should imrove the duplicate issue

**Comment by ricotrevisan** [comment]
Thanks -- I didn't know about import maps. I tried that and it still didn't work (I've updated the original Codepen). 
![CleanShot 2024-06-16 at 08 07 59](https://github.com/esm-dev/esm.sh/assets/2459327/aa0e5b82-0644-446f-a06e-ee23bf03ce55)


While investigating this solution I did notice that I can't load `https://esm.sh/@tiptap/pm` (nor `https://esm.sh/gh/ueberdosis/tiptap@v2.4.0/packages/pm`). Should I be able to load that package?

**Comment by ije** [comment]
i will look into it!

**Comment by jamesbvaughan** [comment]
I'm also interested in more control over transitive dependency versions. I ran into this issue when making use of `globe.gl`.

In case it's useful to see more examples of the issue, you can reproduce it by opening this document in a browser and inspecting the network traffic:
```html
<html>
  <body>
    <script type="module">
      import "https://esm.sh/globe.gl@2.33.0";
    </script>
  </body>
</html>

```
